devices/src/virtio/net.rs

```diff
@@ -489,53 +502,72 @@ impl NetEpollHandler {
 }
 
 impl EpollHandler for NetEpollHandler {
-    fn handle_event(&mut self, device_event: DeviceEventT, _: u32, _: EpollHandlerPayload) {
+    fn handle_event(
+        &mut self,
+        device_event: DeviceEventT,
+        _: u32,
+        _: EpollHandlerPayload,
+    ) -> std::result::Result<(), DeviceErr> {
         match device_event {
+            RX_QUEUE_EVENT => {
+                METRICS.net.rx_queue_event_count.inc();
```

Comment:
I think this block (and the corresponding bocks in the other `match` branches in this file, as well as block.rs and vhost/handle.rs) would look nicer and more rust-y as:
```rust
                self.rx
                    .queue_evt
                    .read()
                    .and_then(|_| {
                        if !self.rx.rate_limiter.is_blocked() {
                            // If the limiter is not blocked, resume the receiving of bytes.
                            // There should be a buffer available now to receive the frame into.
                            self.resume_rx()
                        } else {
                            Ok(())
                        }
                    })
                    .map_err(|e| {
                        error!("Failed to get rx queue event: {:?}", e);
                        METRICS.net.event_fails.inc();
                        DeviceErr::FailedReadingQueue {
                            event_type: "rx queue event",
                            underlying: e,
                        }
                    })
```

**This does not compile** as is; to make it compile, `signal_processed_queue` should return the underlying `sys_util::Error`, which should be wrapped inside a `DeviceError` at a higher level. This is my personal style preference and I won't hold back the changes because of it - it's a suggestion for increasingly clean Rust.

Reply:
Good idea. Done.